src/air/constraints/evaluator.rs

```diff
@@ -108,9 +111,15 @@ impl<'poly, F: IsFFTField, A: AIR + AIR<Field = F>> ConstraintEvaluator<'poly, F
             .collect();
 
         #[cfg(debug_assertions)]
-        for (poly, z) in boundary_polys.iter().zip(boundary_zerofiers.iter()) {
+        for (i, (poly, z)) in boundary_polys
+            .iter()
+            .zip(boundary_zerofiers.iter())
+            .enumerate()
+        {
             let (_, b) = poly.clone().long_division_with_remainder(z);
-            assert_eq!(b, Polynomial::zero());
+            if b != Polynomial::zero() {
+                error!("Expected remainder of boundary poly and boundary zerofier {} to be zero, but it's {:?}", i, b);
```

Comment:
While this is correct, what we really care is if both polynomials are divisible. So we could just say "Boundary poly {i} is not divisible by its zerofier" or something similar. 

Also, we could make this a function receiving the polys and the zerofiers, what do you think? We can add it to de `debug` module.